src/amo/reducers/addonsByAuthors.js

```diff
@@ -135,20 +135,18 @@ export const getAddonsForUsernames = (
   const ids = usernames.map((username) => {
     return state.byUsername[username];
   }).reduce((array, addonIds) => {
-    if (!addonIds || !array) {
-      return null;
-    }
-
-    for (const addonId of addonIds) {
-      if (!array.includes(addonId)) {
-        array.push(addonId);
+    if (addonIds) {
```

Comment:
Do we have a test covering `addonIds` being undefined?

Reply:
Yes there are a couple, including the one I added for this patch.